### PR TITLE
[14.0][IMP] dms: Refactor tests. 

### PR DESCRIPTION
**Pull request by victoralmau**
Refactor tests. 

Please @pedrobaeza could you review it?

@Tecnativa

### CODE REVIEW

File: dms/tests/test_portal.py [hunk 1]
```diff
@@ -14,7 +14,19 @@ def setUp(self):
         self.user_admin = self.env.ref("base.user_admin")
         self.file_text = self.env.ref("dms.file_13_demo")
         self.model_partner = self.env.ref("base.model_res_partner")
-        storage = self.env.ref("dms.storage_attachment_demo")
+        # We need to create a new storage and directory because records may already
+        # exist (e.g. if l10n_en_aeat has been installed).
+        storage = self.env.ref("dms.storage_attachment_demo").copy(
```
Comment (pedrobaeza) [inline dms/tests/test_portal.py]: Don't use demo data

File: dms/tests/test_storage_attachment.py [hunk 2]
```diff
@@ -57,6 +57,9 @@ def setUp(self):
                 ],
             }
         )
+        self.env.ref("base.group_private_addresses").write(
```
Comment (pedrobaeza) [inline dms/tests/test_storage_attachment.py]: Why this? I don't think it's needed.

File: dms/tests/test_portal.py [hunk 3]
```diff
@@ -14,7 +14,19 @@ def setUp(self):
...
+        )
+        self.env["dms.directory"].create(
+            {
+                "name": "Test partners",
+                "is_root_directory": True,
+                "storage_id": storage.id,
+                "res_model": self.model_partner.model,
+            }
+        )
         self.partner = self.env.ref("base.partner_demo_portal")
```
Comment (pedrobaeza) [inline dms/tests/test_portal.py:14]: Don't use demo data

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: dms/tests/test_portal.py (hunk 3 above)
Comment (victoralmau) [reply dms/tests/test_portal.py:14]: Almost all tests are based on demo data and users with specific permissions, I think this should be changed, what do you think?

Comment (pedrobaeza) [reply dms/tests/test_portal.py:14]: Ok

**Comment by victoralmau** [comment]
Changes done and code refactor.

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Pull request merged**